pkg/vendors/mellanox/mellanox.go

```diff
@@ -162,6 +163,15 @@ func (m *mellanoxHelper) MlxResetFW(pciAddresses []string) error {
 func (m *mellanoxHelper) MlxConfigFW(attributesToChange map[string]MlxNic) error {
 	log.Log.Info("mellanox-plugin configFW()")
 	for pciAddr, fwArgs := range attributesToChange {
+		bfMode, err := m.GetMellanoxBlueFieldMode(pciAddr)
+		if err != nil {
+			// NIC is not a DPU or mstconfig failed. It's safe to continue FW configuration
+			log.Log.V(2).Info("mellanox-plugin: configFW(): can't get DPU mode for NIC", "pciAddress", pciAddr)
+		}
+		if bfMode == BluefieldDpu {
+			// Host reboot won't re-load NIC firmware in DPU mode. To apply FW changes power cycle is required or mstfwreset could be used.
```

Comment:
base on this comment should we allow the configuration if the fwreset feature gate is enabled?